exonum-java-binding/core/rust/exonum-java/src/node.rs

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use java_bindings::{Command, Config, EjbCommand, EjbCommandResult};
+
+use java_bindings::exonum::blockchain::{Blockchain, BlockchainBuilder, InstanceCollection};
+use java_bindings::exonum::exonum_merkledb::{Database, RocksDB};
+use java_bindings::exonum::node::{ApiSender, Node, NodeChannel};
+use java_bindings::exonum::runtime::rust::ServiceFactory;
+use std::sync::Arc;
+
+pub fn run_node(command: Command) -> Result<(), failure::Error> {
+    if let EjbCommandResult::EjbRun(config) = command.execute()? {
+        let node = create_node(config)?;
+        node.run()
+    } else {
+        Ok(())
+    }
+}
+
+fn create_node(config: Config) -> Result<Node, failure::Error> {
+    let channel = NodeChannel::new(&config.run_config.node_config.mempool.events_pool_capacity);
+    let blockchain = create_blockchain(&config, &channel)?;
+
+    let node_config = config.run_config.node_config;
+    let node_config_path = config
+        .run_config
+        .node_config_path
+        .to_str()
+        .expect("Cannot convert node_config_path to String")
+        .to_owned();
+
+    Ok(Node::with_blockchain(
+        blockchain,
+        channel,
+        node_config,
+        Some(node_config_path),
+    ))
+}
+
+fn create_blockchain(config: &Config, channel: &NodeChannel) -> Result<Blockchain, failure::Error> {
+    let node_config = &config.run_config.node_config;
+    let service_factories = standard_exonum_service_factories();
+    let database = create_database(config)?;
+    let keypair = node_config.service_keypair();
+    let api_sender = ApiSender::new(channel.api_requests.0.clone());
+    let internal_requests = channel.internal_requests.0.clone();
+
+    BlockchainBuilder::new(database, node_config.consensus.clone(), keypair)
+        // TODO: add Java runtime
+        .with_rust_runtime(service_factories.into_iter().map(InstanceCollection::new))
```

Comment:
InstanceCollection? We don't seem to have instances unless started, do we?

Reply:
Well, it's Core's term for service artifacts (?). It feels weird, yeah.